### PR TITLE
Daniil/screen when day started

### PR DESCRIPTION
**Pull request by Xirisss**
screen when day started

### CODE REVIEW

File: src/components/NewActivityList.js [hunk 1]
```diff
@@ -24,20 +30,52 @@ class NewActivityList extends Component {
     firebase.database().ref('/users/' + userId + '/settings').once('value').then(snapshot => {
```
Comment (Stas-Buzunko) [inline src/components/NewActivityList.js]: 1) i'd like to see you checking in snapshot is not null before using it
if (snapshot.val() !== null) {

2) let's decide to use it like this
```
const object (you can name it as you want) = snapshot.val()
if (object !== null) {
 doSomething
}
```

File: src/components/NewActivityList.js [hunk 2]
```diff
@@ -24,20 +30,52 @@ class NewActivityList extends Component {
...
       const currentRate = (snapshot.val().currentRate * 0.4)
```
Comment (Stas-Buzunko) [inline src/components/NewActivityList.js]: why not name it money to use shorthand syntax?

File: src/components/NewActivityList.js [hunk 3]
```diff
@@ -24,20 +30,52 @@ class NewActivityList extends Component {
...
       this.setState({
-        money: currentRate
+        money: currentRate,
+        how_long: snapshot.val().preference,
+        workspace: snapshot.val().workspace,
```
Comment (Stas-Buzunko) [inline src/components/NewActivityList.js]: when u have the same key and value it feels good to use spread + shorthand syntax
```
const { workspace } = object
this.setState({ workspace })
```
it's the same for other keys

File: src/components/NewActivityList.js [hunk 4]
```diff
@@ -24,20 +30,52 @@ class NewActivityList extends Component {
...
+        togglKey: snapshot.val().togglKey,
+        email: snapshot.val().email,
+        user_ids: snapshot.val().user_ids
       })
     })
+
     firebase.database().ref('/users/' + userId + '/challenge').once('value').then(snapshot => {
       if (snapshot.val().startedAt + 86400000 > moment()) {
```
Comment (Stas-Buzunko) [inline src/components/NewActivityList.js]: what if snapshot.val() is null?
startedAt from null?

Comment (Stas-Buzunko) [reply src/components/NewActivityList.js]: add check for null

File: src/components/NewActivityList.js [hunk 5]
```diff
@@ -24,20 +30,52 @@ class NewActivityList extends Component {
...
         this.setState({
           hours: snapshot.val().hours,
```
Comment (Stas-Buzunko) [inline src/components/NewActivityList.js]: spread + shorthand

File: src/components/NewActivityList.js [hunk 6]
```diff
@@ -24,20 +30,52 @@ class NewActivityList extends Component {
...
+      },
+      url: 'https://toggl.com/reports/api/v2/details',
+      params: {
+        user_agent: this.state.email,
+        workspace_id: this.state.workspace,
+        user_ids: this.state.user_ids,
+        since: time
       }
     })
+    .then((response) => {
```
Comment (Stas-Buzunko) [inline src/components/NewActivityList.js]: can be refactored to
```
.then(response => this.setState({ time: response.data.total_grand  }))
```

File: src/components/NewActivityList.js [hunk 7]
```diff
@@ -49,24 +87,50 @@ class NewActivityList extends Component {
...
+        workspace_id: this.state.workspace,
+        user_ids: this.state.user_ids,
+        since: time
+      }
+    })
+    .then((response) => {
+      console.log(response)
+      const timeWorkedBefore = response.data.total_grand
+      this.setState({
+        timeWorkedBefore: timeWorkedBefore
```
Comment (Stas-Buzunko) [inline src/components/NewActivityList.js]: key = value => shorthand

File: src/components/NewActivityList.js [hunk 8]
```diff
@@ -76,6 +140,7 @@ class NewActivityList extends Component {
       money,
       how_long,
       userUid,
+      timeWorkedBefore,
```
Comment (Stas-Buzunko) [inline src/components/NewActivityList.js]: let's refactor these 2 requests to firebase
there are 2 ways
1) you want to send 2nd request after first is saved - use .then() then
2) you want to save at the same time, then we can do it in 1 request, 
https://firebase.google.com/docs/database/web/read-and-write#updating_or_deleting_data

File: src/pages/SettingsPage.js [hunk 9]
```diff
@@ -20,6 +22,8 @@ class Settings extends Component {
       this.setState({
         togglKey: snapshot.val().togglKey,
         workspace: snapshot.val().workspace,
+        email: snapshot.val().email,
```
Comment (Stas-Buzunko) [inline src/pages/SettingsPage.js]: check for snapshot !== null + spread + shorthand

**Review by Stas-Buzunko — CHANGES_REQUESTED** [review]

File: src/components/CountDown.js [hunk 10]
```diff
@@ -18,33 +19,17 @@ class CountDown extends Component {
...
-      this.tillMidnight()
-    } else {
-      this.h24()
-    }
-  }
-
-  tillMidnight() {
-    const startedAt = this.props.startedAt;
-    if (this.props.startedAt && this.props.hours) {
+      const startedAt = this.props.startedAt;
```
Comment (Stas-Buzunko) [inline src/components/CountDown.js]: spread

File: src/components/NewActivityList.js [hunk 11]
```diff
@@ -26,11 +26,15 @@ class NewActivityList extends Component {
   }
 
   componentDidMount() {
-    const userId = firebase.auth().currentUser.uid;
-    firebase.database().ref('/users/' + userId + '/settings').once('value').then(snapshot => {
-      const currentRate = (snapshot.val().currentRate * 0.4)
+    const { uid } = this.props.user
+    firebase.database().ref('/users/' + uid + '/settings').once('value').then(snapshot => {
+      if (snapshot.val().currentRate) {
```
Comment (Stas-Buzunko) [inline src/components/NewActivityList.js]: if snapshot.val() is null, does null.currentRate will break the app?
check what happens when you try to use key from null

Comment (Stas-Buzunko) [reply src/components/NewActivityList.js]: .then() can be refactored to 1-3 lines

File: src/pages/SettingsPage.js [hunk 12]
```diff
@@ -17,8 +17,8 @@ class Settings extends Component {
   }
 
   componentDidMount() {
-    const userId = firebase.auth().currentUser.uid;
-    firebase.database().ref('/users/' + userId + '/settings').once('value').then(snapshot => 
+    const { uid } = this.props.user
```
Comment (Stas-Buzunko) [inline src/pages/SettingsPage.js]: :+1:

**Review by Stas-Buzunko — CHANGES_REQUESTED** [review]

File: src/components/NewActivityList.js (hunk 1 above)
Comment (Xirisss) [reply src/components/NewActivityList.js]: done

File: src/components/NewActivityList.js (hunk 2 above)
Comment (Xirisss) [reply src/components/NewActivityList.js]: done

File: src/components/NewActivityList.js (hunk 4 above)
Comment (Xirisss) [reply src/components/NewActivityList.js]: done

File: src/components/NewActivityList.js (hunk 6 above)
Comment (Xirisss) [reply src/components/NewActivityList.js]: done

File: src/components/NewActivityList.js (hunk 3 above)
Comment (Xirisss) [reply src/components/NewActivityList.js]: done

File: src/components/NewActivityList.js (hunk 5 above)
Comment (Xirisss) [reply src/components/NewActivityList.js]: done

File: src/components/NewActivityList.js (hunk 7 above)
Comment (Xirisss) [reply src/components/NewActivityList.js]: done

File: src/components/NewActivityList.js (hunk 8 above)
Comment (Xirisss) [reply src/components/NewActivityList.js]: done

File: src/pages/SettingsPage.js (hunk 9 above)
Comment (Xirisss) [reply src/pages/SettingsPage.js]: done

File: src/components/CountDown.js (hunk 10 above)
Comment (Xirisss) [reply src/components/CountDown.js]: done

File: src/components/NewActivityList.js (hunk 11 above)
Comment (Xirisss) [reply src/components/NewActivityList.js]: added:
const snapshotval = snapshot.val()
if if (snapshotval !== null) {
   ...
}

File: src/components/CountDown.js [hunk 13]
```diff
@@ -0,0 +1,54 @@
...
+  }
+
+  renderTime() {
+    if (this.props.how_long) {
+      const { startedAt } = this.props;
+      const left = Number(startedAt) + 24 * 60 * 60 * 1000 - Date.now();
+      this.calculationTime(left)
+    } else {
+      const left = 24 * 60 * 60 * 1000 - Date.now() % (24 * 60 * 60 * 1000);
+      this.calculationTime(left)
```
Comment (Stas-Buzunko) [inline src/components/CountDown.js]: you call calculationTime in both cases, how about only setting "left" in if and then call calculationTime?

**Review by Stas-Buzunko — APPROVED** [review]

File: src/components/CountDown.js (hunk 13 above)
Comment (Xirisss) [reply src/components/CountDown.js]: done

**Pull request merged**